### ISSUE TITLE
Paths of Fuzz Items contains ".0."

### ISSUE THREAD

**Issue by thatjimmi** [apideck-libraries/portman#335]
The Fuzzing feature uses failing paths with ".0." in some of the requests that is made.

Example: [required payload.Workorder.0.door]

What can the issue be? Is the nesting failing or something?

**Comment by thim81** [comment]
Hi @thatjimmi 

Would it be possible to share your openapi spec and your Portman config, so we can have a closer look at the input and generated output?

**Comment by thatjimmi** [comment]
Hi @thim81 
Here is a link: https://drive.google.com/file/d/1lP-asLgMmZ-eozfLMNtM4pbM5ZKhqhuD/view?usp=sharing

**Comment by thatjimmi** [comment]
I think it has to do with "allOf"

**Comment by thim81** [comment]
@thatjimmi Thanks for the files and tip, It will help with the investigation.

**Comment by yelob** [comment]
Just to chime in here, I'm also finding an issue with using "allOf" to pull in a base schema. Portman generates the variation tests, but doesn't properly set the body of the requests.

Copying the properties of the referenced schema causes the tests to be generated correctly.

I've attached an example schema and the resulting postman tests to help you diagnose.
[temp-schema.txt](https://github.com/apideck-libraries/portman/files/8825844/temp-schema.txt)
[postman.txt](https://github.com/apideck-libraries/portman/files/8825847/postman.txt)

**Comment by thim81** [comment]
@thatjimmi @yelob 

We were able to reproduce the issue.
Just to be sure: @yelob the items not crossed out in the screenshot are the ones with the unwanted 0. path?

![2022-06-07 at 09 54 10](https://user-images.githubusercontent.com/952446/172326907-6bd75aa1-e6eb-40c8-8737-26ffbf83d19a.png)

**Comment by thim81** [comment]
@yelob, In the PR, that seems to fix it, this how the Fuzzing variation looks in Postman.
Can you review the screenshot result and see if this would match with the expected behaviour?

![2022-06-07 at 11 42 57](https://user-images.githubusercontent.com/952446/172349860-bfc00761-6c20-4b1d-aaf6-7beed9c81170.png)

**Comment by thatjimmi** [comment]
Really nice! I see that @yelob's example seems fixed, but mine is still is not working.

![Skærmbillede 2022-06-07 kl  14 06 34](https://user-images.githubusercontent.com/52856973/172375740-feea2037-9716-4f68-b391-67b2001f8b0a.png)
![Skærmbillede 2022-06-07 kl  14 08 00](https://user-images.githubusercontent.com/52856973/172375745-0bfccd0c-ccd0-4b49-b0f6-05347e4568e6.png)

**Comment by thatjimmi** [comment]
Yeah, I have validated the above example from yelop using the PR and validated that it worked. 

Hmm, I'm not sure what you mean by missing $references. You can't get it to work in general?

**Comment by yelob** [comment]
> @yelob, In the upcoming PR, that seems to fix it, this is how the Fuzzing variation looks in Postman. Can you review the screenshot result and see if this would match with the expected behaviour?
> 
> ![2022-06-07 at 11 42 57](https://user-images.githubusercontent.com/952446/172349860-bfc00761-6c20-4b1d-aaf6-7beed9c81170.png)

Yep, that's exactly what I was looking for, thanks for getting to this so quickly!

**Comment by thim81** [comment]
@yelob thanks for the confirmation. I'm hoping that we have solved the issue of @thatjimmi in the same go, since that is the final validation I'm looking for, before releasing a new Portman version.

**Comment by thim81** [comment]
Could you share the OAS file here again, by adding the `.txt` extension?

**Comment by thatjimmi** [comment]
[openapi.txt](https://github.com/apideck-libraries/portman/files/8854066/openapi.txt)

**Comment by thim81** [comment]
@thatjimmi When focussing on 1 test, it seems that only the title is still and the response body is not properly altered?

![2022-06-07 at 17 08 41@2x](https://user-images.githubusercontent.com/952446/172415686-bbf61861-250b-40b4-a2a4-f91fec7dd4a1.png)

The expected situation would have been:
- title: `commerce Iceorder[Bad][required payload.Iceorder.0.addressAsText]`
- req body: 
```json
{
    "payload": {
        "Iceorder": {
            "address": {
                "cityName": "est in cupidatat",
                "countryCode": "nostrud dolore",
                "darId": "nulla non minim ad",
                "door": "velit Lorem aliquip qui",
                "floor": -32589445,
                "localName": "Lorem exercitation",
                "municipialityName": "aliquip cupidatat eu reprehenderit",
                "postalCode": 53323675,
                "region": "magna in",
                "streetName": "non",
                "streetNumber": ""
            }
        }
    }
}
```

**Comment by thatjimmi** [comment]
The expected path would be "payload.Iceorder.address.addressAsText" if it's normal iteration of an object, right?

So the ".0." is the error and should be "address"

**Comment by thim81** [comment]
@thatjimmi a screenshot of the fixed generated allOf items for the nested "Address" with the required "addressText"

<img width="847" alt="2022-06-07 at 21 20 14@2x" src="https://user-images.githubusercontent.com/952446/172464803-7ea6da98-2ece-4d81-9876-1af8fbc890ba.png">

I think, this is what would be expected? Correct?

**Comment by thim81** [comment]
@thatjimmi We have been trying get this issue sorted out.

We keep running into strange behaviour when using "minLength" for an object.

```json
"address": {
                        "minLength": 1,
                        "allOf": [
                            {
                                "$ref": "#/components/schemas/Address"
                            }
                        ]
                    }
```

Did you find an OpenAPI example or in the specification that this is properly supported?

**Comment by thatjimmi** [comment]
We have received this specification from one of our partners. I've talked with my colleagues and concluded that a "minLength" of an object doesn't make much sense.  So it does what we need it to do :)

Great work!

**Comment by thim81** [comment]
@thatjimmi @yelob We just release Portman v1.17, which includes the improvements as mentioned in this item.

I'm going to close this item, but would something pop up, feel free to create a new item.

**Comment by fellnerse** [comment]
Awesome I had the same problem and updating fixed it!
One thing that changed so is that now only one variation in total is generated:

I have a parent `oneOf` that wraps multiple items. Before the update all required fields of the children got this weird `0.` prefix. But now they are just gone. My parent schema looks like this:

```yaml
Customer:
  required:
    - customerType
  oneOf:
    - $ref: "#/components/schemas/Customer1"
    - $ref: "#/components/schemas/Customer2"
  discriminator:
    propertyName: customerType
    mapping:
      customer1: "#/components/schemas/Customer1"
      customer2: "#/components/schemas/Customer2"
```

I get the variation for customerType now but nothing else. Only I also add the required fields of the children here it works again.
Before the update the generated postman file was around 16k lines now its only 600.

Kind regards!

**Comment by thim81** [comment]
@fellnerse I'll try your provided example and see how Portman handles this.
